client/src/pages/EditProfilePage/EditProfilePage.tsx

```diff
@@ -21,32 +22,167 @@ const EditProfilePage = () => {
   const fileInputRef = useRef<HTMLInputElement>(null);
 
   const [formData, setFormData] = useState({
+    cohort: "",
     email: "",
+    firstName: "",
+    linkedInProfile: "",
+    gitHubProfile: "",
+    nickName: "",
     personalBio: "",
+    skills: [] as string[],
+    specializations: [] as string[],
+    careerInformation: {
+      currentPosition: {
+        title: "",
+        company: "",
+      },
+    },
+    socialMediaLinks: {
+      twitter: "",
+      blog: "",
+    },
+    availabilityForNetworking: false,
   });
 
+ 
+
   const [file, setFile] = useState<File | null>(null);
 
   useEffect(() => {
     if (userID) dispatch(fetchUserProfile(userID as string));
   }, [dispatch]);
+
   useEffect(() => {
     if (profile) {
       setFormData({
+        cohort: profile.cohort || "",
         email: profile.email || "",
+        firstName: profile.firstName || "",
+        linkedInProfile: profile.linkedInProfile || "",
+        gitHubProfile: profile.gitHubProfile || "",
+        nickName: profile.nickName || "",
         personalBio: profile.personalBio || "",
+        skills: profile.skills || [],
+        specializations: profile.specializations || [],
+        careerInformation: {
+          currentPosition: {
+            title: profile.careerInformation?.currentPosition?.title || "",
+            company: profile.careerInformation?.currentPosition?.company || "",
+          },
+        },
+        socialMediaLinks: {
+          twitter: profile.socialMediaLinks?.twitter || "",
+          blog: profile.socialMediaLinks?.blog || "",
+        },
+        availabilityForNetworking: profile.availabilityForNetworking || false,
       });
     }
   }, [profile]);
 
-  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+  const handleChange = (
```

Comment:
All the event handlers have similar functionality. Can we find a way to abstract a bit and not have to declare a new event handler for each input field?

Reply:
Yes, good call! I was able to consolidate the nested event handlers into a single function.